pkg/minikube/image/image.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/docker/docker/client"
+	"github.com/golang/glog"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/daemon"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"k8s.io/minikube/pkg/minikube/constants"
+)
+
+// DigestByLocalDaemon uses client by docker lib
+func DigestByLocalDaemon(imgClient *client.Client, imgName string) string {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+	img, _, err := imgClient.ImageInspectWithRaw(ctx, imgName)
+	if err != nil && !client.IsErrImageNotFound(err) {
+		glog.Infof("couldn't find image digest %s from local daemon: %v ", imgName, err)
+		return ""
+	}
+	return img.ID
+}
+
+// DigestByRetrieve gets image digest uses go-containerregistry lib
+// which is 4s slower thabn local daemon per lookup https://github.com/google/go-containerregistry/issues/627
+func DigestByRetrieve(imgName string) string {
```

Comment:
I think `DigestGoContainerregistry` might be a clearer name, but no worries if you'd rather keep it as is!